doc/source/operations/cloudkitty.rst

```diff
@@ -0,0 +1,125 @@
+
+  kolla_enable_cloudkitty: true
+  # Explicitly disable influxdb as we are using OpenSearch as the CloudKitty backend
+  kolla_enable_influxdb: false
+
+Use Prometheus as the collector and fetcher backends, and Elasticsearch as the
+storage backend. Note that this is patched in our fork of CloudKitty to also
+work with OpenSearch. Proper support is still pending in Kolla-Ansible `here
+<https://review.opendev.org/c/openstack/kolla-ansible/+/898555>`__. If you have
+TLS enabled, you will also need to allow insecure connections for Prometheus
+and Elasticsearch. Set the following in ``kolla/globals.yml``:
+
+.. code-block:: console
+
+  cloudkitty_collector_backend: prometheus
+  cloudkitty_fetcher_backend: prometheus
+  cloudkitty_storage_backend: elasticsearch
+  cloudkitty_elasticsearch_insecure_connections: true
+  cloudkitty_prometheus_insecure_connections: true
+
+The default collection period is one hour, which is likely too long for your
+system. CloudKitty will charge for the entire period if any usage is seen
+within this timeframe. This means that even just one minute will be counted as
+an hour's usage. It is recommended to change this to a lower number, such as
+ten minutes. When using Prometheus as the collector, you need to change the
+scope_key to match the metrics provided by the Prometheus OpenStack Exporter.
+Set the following in ``cloudkitty.conf``:
+
+.. code-block:: console
+
+  [collect]
+  scope_key = tenant_id
+  period = 600
+
+You will need to configure which metrics CloudKitty should track. This example
+will track for flavors and volumes, set in
+``kolla/config/cloudkitty/metrics.yml``:
+
+.. code-block:: console
+
+  metrics:
+    openstack_nova_server_status:
+        alt_name: instance
+        groupby:
+        - uuid
+        - user_id
+        - tenant_id
+        metadata:
+        - flavor_id
+        - name
+        mutate: MAP
+        mutate_map:
+        0.0: 1.0  # ACTIVE
+        11.0: 1.0 # SHUTOFF
+        12.0: 1.0 # SUSPENDED
+        16.0: 1.0 # PAUSED
+        unit: instance
+    openstack_cinder_limits_volume_used_gb:
+        alt_name: storage
+        unit: GiB
+        groupby:
+        - tenant_id
+
+TODO: Should we explain how to handle OpenStack exporter metrics relabelling,
+for the case where a system may have had Monasca deployed in the past?
+
+Post-configuration with openstack-config
+========================================
+
+This is an example `openstack-config
+<https://github.com/stackhpc/openstack-config>`__ setup to create mappings for
+the metrics configured above. Note that the costs are scaled for the ten-minute
+collection period, e.g. a flavor with 1 VCPU will cost 1 unit per hour.
+
+.. code-block:: yaml
+
+  # Map flavors based on VCPUs
+  openstack_ratings_hashmap_field_mappings:
+    - service: instance
+        name: flavor_id
+        mappings:
+        - value: '1' # tiny compute flavor (1 vcpu) with an OpenStack flavor ID of 1
+            cost: 0.1666666666666666
```

Comment:
:nauseated_face:

Reply:
Perhaps jinja can template this, don't think it's really an issue though.